dragonmapper/html.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+"""Formatting Chinese into HTML with dragonmapper's functions"""
+
+
+from __future__ import unicode_literals
+from dragonmapper import hanzi
+from dragonmapper import transcriptions as trans
+import zhon
+from zhon import pinyin
+from zhon import zhuyin
+
+
+"""See recomended CSS style: DRAGONMAPPER_DIR/style.css"""
+
+
+_indentation = 0
+_line_html = ''
+_puctuation = tuple(zhon.hanzi.punctuation + zhon.pinyin.punctuation)
+_tones_marks = ['¯', 'ˊ', 'ˇ', 'ˋ', '˙', '1', '2', '3', '4', '5']
+
+
+def _identify(s):
+
+    """
+    Returns string of text type for HTML/CSS.
+
+    *s* is the string to identify.
+    """
+
+    if hanzi.has_chinese(s):
+        return "hanzi"
+    elif s == "":
+        return "unknown"
+    elif s in _puctuation:
+        return "punct"
+    elif s in _tones_marks:
+        return "tone-mark"
+    else:
+        c = trans.identify(s)
+        if c == trans.ZHUYIN:
+            return "zhuyin"
+        elif c == trans.PINYIN:
+            return "pinyin"
+        elif c == trans.UNKNOWN:
+            return "unknown"
+
+
+def _stackify(s):
+
+    """
+    Stack string for HTML formatting on the left and right of characters.
+
+    *s* is the string to "stackify".
+    """
+
+    return "<br />".join(list(s))
+
+
+def _html_add(s, tabs=0):
+
+    """
+    Wrapper for _line_html+="..."
+
+    *s* is what to add to the html string.
+    *tabs* specifies the identation intensity (in tabs).
+    """
+
+    global _line_html
+    _line_html += (("\n")+("\t"*(tabs+_indentation)))+s
+
+
+def _split_punct(s):
+
+    """
+    Internal function for spliting by punctuation only for HTML formatting.
+
+    *s* specifies the list to preform this action on.
+    """
+
+    temp = []
+
+    s = s.split(' ')
+
+    for c in s:
+        full_char_temp = ""
+        for pc in c:
+            if (pc in zhuyin.characters or
+                    pc in pinyin.vowels or
+                    pc in pinyin.consonants or
+                    pc in _tones_marks or
+                    pc in [1, 2, 3, 4, 5]):
```

Comment:
`[1, 2, 3, 4, 5]` is unneeded.

Try running `'1' in [1, 2, 3, 4, 5]` to see what I mean. `'1'` is in `_tones_marks`.